File: Guanabara - PHP/exercicios-PHPOO/Aula04/Caneta.php
<?php
class Caneta {
    //declarando atributos
    private $modelo;    
    private $ponta;
    private $cor;
    private $tampada;

    //método construtor
    public function __construct()   
    {
        $this->cor = "Azul";
        $this->ponta = 0.5;    
        $this->tampar();    //quando uma nova classe Caneta é instanciada, ela já começa com cor Azul, ponta 0.5 e tampada
    }
    /*public function Caneta($m, $c, $p){  //ou podemos criar o método construtor usando o mesmo nome da sua classe
        $this->modelo = $m;
        $this->cor = $c;
        $this->ponta = $p;
        $this->tampar();
    }*/

    //métodos acessores
    public function getModelo() {
        return $this->modelo;
    }
    public function setModelo($mod) {
        $this->modelo = $mod;
    }
    public function getPonta() {
        return $this->ponta;
    }
    public function setPonta($pon) {
        $this->ponta = $pon;
    }

    //métodos públicos
    public function tampar(){
        $this->tampada = true;
    }
}

File: Java/Meus exercícios POO/Aula04/src/main/java/cursoemvideo/Caneta.java
package cursoemvideo;
public class Caneta {
//declarando as variáveis
    public String modelo;   
    private float ponta;
    private String cor;
    private int tamanho;
    private boolean tampada;

//Este é o método abaixo é o CONSTRUTOR, ele deve ter o mesmo nome de sua classe.    
    public Caneta(String m, float p, int t){ //Esse método contrutor também receberá 3 parâmetros: Modelo, Ponta e Tamanho.
        this.cor = "Azul";  //a caneta já será azul quando criada
        
        this.setModelo(m);  //tanto posso atribuir valor pela método acessor SETTER (+seguro)
        this.setPonta(p);   //quanto posso atribuir valor diretamente
        this.tamanho = t;   //como não criei um método acessor para `tamanho` então só posso atribuir valor diretamente
        this.tampar();      //o objeto caneta já nasce tampado
    }
    
    public String getModelo(){  //é sem parâmetro mas retorna valor
        return this.modelo;
    }
    
    public void setModelo(String m){    //é VOID pois não retorna nada
        this.modelo = m;
    }
    
    public float getPonta(){
        return this.ponta;
    }
    
    public void setPonta (float p){
        this.ponta = p;
    }
    
    public void status(){
        System.out.println("SOBRE A CANETA:");
        System.out.println("Modelo: " + this.modelo);
        System.out.println("Ponta: " + this.ponta);
        System.out.println("Cor: " + this.cor);
        System.out.println("Tampada? " + this.tampada);
    }
    
    public void tampar(){
        this.tampada = true;
    }
    
    public void destampada(){
        this.tampada = false;
    }
}
